### PR TITLE
Add Cmd interface for extending command execution

### PR DESCRIPTION
**Pull request by marineam**
Depends on https://github.com/coreos/mantle/pull/10

### CODE REVIEW

File: util/exec.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	// Simplified wrapper for Process.Kill + Wait
+	Kill() error
+}
+
+// Basic Cmd implementation based on exec.Cmd
+type ExecCmd struct {
+	*exec.Cmd
+}
+
+func NewCommand(name string, arg ...string) Cmd {
```
Comment (crawford) [inline util/exec.go]: I think returning `*ExecCmd` here is a bit more useful.

File: platform/local/nsexec.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	"github.com/coreos/mantle/util"
+)
+
+type NsCmd struct {
+	util.ExecCmd
+	NsHandle netns.NsHandle
+}
+
+func NewNsCommand(ns netns.NsHandle, name string, arg ...string) util.Cmd {
```
Comment (crawford) [inline platform/local/nsexec.go]: return `*NsCmd`

File: platform/local/nsexec.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	return &NsCmd{
+		ExecCmd:  util.ExecCmd{exec.Command(name, arg...)},
+		NsHandle: ns,
+	}
+}
+
+func (cmd *NsCmd) CombinedOutput() ([]byte, error) {
```
Comment (crawford) [inline platform/local/nsexec.go]: You could probably encapsulate this pattern in a helper function and reduce this to `return execWithinNS(cmd.ExecCmd.CombinedOutput())`.

Comment (marineam) [inline platform/local/nsexec.go]: Except each of the functions have a different set of return values to pass through.

**Comment by marineam** [comment]
@crawford updated

File: platform/local/nsexec.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	}
+	defer nsExit()
+
+	return cmd.ExecCmd.CombinedOutput()
+}
+
+func (cmd *NsCmd) Output() ([]byte, error) {
+	nsExit, err := NsEnter(cmd.NsHandle)
+	if err != nil {
+		return []byte{}, err
```
Comment (crawford) [inline platform/local/nsexec.go]: Is it intentional to return an empty slice here, but a nil from `CombinedOutput`?

Comment (marineam) [inline platform/local/nsexec.go]: No, just plain inconsistency. Which is preferable?

**Comment by crawford** [comment]
LGTM

**Comment by pbx0** [comment]
LGTM

**Pull request merged**